### PR TITLE
Added arch/arm && docker rebased to quay.io

### PR DESCRIPTION
**Pull request by shuangjj**
- All arm docker build files in arch/arm
- The current suported docker build for arm: *[base, data, ipfs, btcd, ubuntu,
tools, node, gulp, commonform, keys]*
- Need to refine and test build script on arm

### CODE REVIEW

**Comment by AFDudley** [comment]
This work looks mostly correct to me, but we need to re-evaluate the structure of this code before we can take a PR this large.

File: arch/arm/cloud/setup/setup.sh [hunk 1]
```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/env bash
```
Comment (zramsay) [inline arch/arm/cloud/setup/setup.sh]: we can kill this file completely

File: arch/arm/docker/bigchaindb/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+FROM python:3.5
```
Comment (zramsay) [inline arch/arm/docker/bigchaindb/Dockerfile]: remove file

File: arch/arm/docker/bigchaindb/bigchaindb.config [hunk 3]
```diff
@@ -0,0 +1 @@
+{"database": {"host": "rethinkdb", "name": "bigchain", "port": 28015}, "keypair": {"private": "BvGagGiTpytjdM5pZmvettvh7qPfVYbhsFRobEwxF6Fz", "public": "gDe65yYUkSZMUWkqk3NZbgjBJZTp6UnVWCx576XcCs1W"}, "keyring": []}
```
Comment (zramsay) [inline arch/arm/docker/bigchaindb/bigchaindb.config]: rm file

File: arch/arm/docker/bitcoinclassic/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,51 @@
+FROM eris4iot/tools:armhf
```
Comment (zramsay) [inline arch/arm/docker/bitcoinclassic/Dockerfile]: quay.io...

File: arch/arm/docker/bitcoincore/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,43 @@
+FROM eris4iot/tools:armhf
```
Comment (zramsay) [inline arch/arm/docker/bitcoincore/Dockerfile]: ibid

File: arch/arm/docker/embark_base/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,12 @@
+FROM eris4iot/gulp:armhf
```
Comment (zramsay) [inline arch/arm/docker/embark_base/Dockerfile]: rm file

File: arch/arm/docker/eth/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,33 @@
+FROM eris4iot/tools:armhf
```
Comment (zramsay) [inline arch/arm/docker/eth/Dockerfile]: rm file: s/eth/geth has happened in #develop in eris-cli

File: arch/arm/docker/eth/new_account.sh [hunk 8]
```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
```
Comment (zramsay) [inline arch/arm/docker/eth/new_account.sh]: rm file

File: arch/arm/docker/eth/start.sh [hunk 9]
```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env bash
```
Comment (zramsay) [inline arch/arm/docker/eth/start.sh]: ibid

File: arch/arm/docker/geth/Dockerfile [hunk 10]
```diff
@@ -0,0 +1,33 @@
+FROM eris4iot/tools:armhf
```
Comment (zramsay) [inline arch/arm/docker/geth/Dockerfile]: quay...

File: arch/arm/docker/gulp/Dockerfile [hunk 11]
```diff
@@ -0,0 +1,10 @@
+FROM eris4iot/node:armhf
```
Comment (zramsay) [inline arch/arm/docker/gulp/Dockerfile]: rm file

File: arch/arm/docker/ipfs/Dockerfile [hunk 12]
```diff
@@ -0,0 +1,66 @@
+FROM eris4iot/base:armhf
```
Comment (zramsay) [inline arch/arm/docker/ipfs/Dockerfile]: qwuay

File: arch/arm/docker/keys/build.sh [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+start=`pwd`
+name=keys
+eris_name=eris-$name
+arch=arm
+repo=https://github.com/eris-ltd/$eris_name
+
+cd /tmp
+git clone $repo $name
+cd $name
+docker build --no-cache -t quay.io/eris/$name:$arch -f arch/arm/docker/Dockerfile . 1>/dev/null
```
Comment (zramsay) [inline arch/arm/docker/keys/build.sh]: :+1:

File: arch/arm/docker/node/Dockerfile [hunk 14]
```diff
@@ -0,0 +1,8 @@
+FROM eris4iot/tools:armhf
```
Comment (zramsay) [inline arch/arm/docker/node/Dockerfile]: quay

File: arch/arm/docker/openbazaar-client/Dockerfile [hunk 15]
```diff
@@ -0,0 +1,20 @@
+FROM eris4iot/node:armhf
```
Comment (zramsay) [inline arch/arm/docker/openbazaar-client/Dockerfile]: rm file

File: arch/arm/docker/openbazaar-server/Dockerfile [hunk 16]
```diff
@@ -0,0 +1,34 @@
+FROM eris4iot/tools:armhf
```
Comment (zramsay) [inline arch/arm/docker/openbazaar-server/Dockerfile]: rm file

File: arch/arm/docker/parity/Dockerfile [hunk 17]
```diff
@@ -0,0 +1,24 @@
+FROM eris4iot/tools:armhf
```
Comment (zramsay) [inline arch/arm/docker/parity/Dockerfile]: rm file

File: arch/arm/docker/sockets/Dockerfile [hunk 18]
```diff
@@ -0,0 +1,10 @@
+FROM quay.io/eris/base
```
Comment (zramsay) [inline arch/arm/docker/sockets/Dockerfile]: rm file

File: arch/arm/docker/sunit_base/Dockerfile [hunk 19]
```diff
@@ -0,0 +1,12 @@
+FROM eris4iot/gulp:armhf
```
Comment (zramsay) [inline arch/arm/docker/sunit_base/Dockerfile]: rm file

File: arch/arm/docker/tendermint/Dockerfile [hunk 20]
```diff
@@ -0,0 +1,31 @@
+# Pull base image.
+FROM quay.io/eris/base
```
Comment (zramsay) [inline arch/arm/docker/tendermint/Dockerfile]: ...base:arm ... ?

File: arch/arm/docker/zcash/Dockerfile [hunk 21]
```diff
@@ -0,0 +1,62 @@
+FROM eris4iot/tools:armhf
```
Comment (zramsay) [inline arch/arm/docker/zcash/Dockerfile]: quay

File: arch/arm/docker/build.sh [hunk 22]
```diff
@@ -0,0 +1,107 @@
...
+
+tobuild=(
+  "base"
+  "data"
+  "ipfs"
+  "btcd"
+  "ubuntu"
+  "tools"
+  ##"eth"
+  ##"geth"
```
Comment (zramsay) [inline arch/arm/docker/build.sh]: we should have geth on arm for sure

File: arch/arm/cloud/setup/setup.sh [hunk 23]
```diff
@@ -0,0 +1,202 @@
...
+echo "Version Information"
+echo
+go version
+echo
+docker version
+echo
+echo
+echo "Building eris."
+pre_dir=$pwd
+go get -d github.com/shuangjj/eris-cli/cmd/eris
```
Comment (zramsay) [inline arch/arm/cloud/setup/setup.sh]: eris-ltd

**Pull request merged**